Source/DafnyLanguageServer/Language/SyntaxTreeVisitor.cs

```diff
@@ -506,6 +528,17 @@ public abstract class SyntaxTreeVisitor {
     }
 
     public virtual void Visit(ComprehensionExpr comprehensionExpression) {
+      foreach (var boundVar in comprehensionExpression.BoundVars) {
+        Visit(boundVar);
+      }
+      VisitNullableAttributes(comprehensionExpression.Attributes);
+      VisitNullableExpression(comprehensionExpression.Range);
+      if (comprehensionExpression is not SetComprehension setComp || !setComp.TermIsImplicit) {
```

Comment:
These type-checks will be unnecessary when creating distinct overloads.
Nevertheless, you may create a common `private` method to handle the common parts of `ComprehensionExpr`.

Reply:
I created distinct overloads now.